wordpress.org/public_html/wp-content/plugins/wporg-gp-translation-suggestions/js/translation-suggestions.js

```diff
@@ -288,8 +300,90 @@
 			original();
 
 			$( $gp.editor.table )
-				.on( 'click', '.translation-suggestion', copySuggestion );
+				.on( 'click', '.translation-suggestion', copySuggestion )
+				.on( 'click', 'button.translation-actions__save', saveExternalSuggestions )
+				.on( 'click', '.translation-suggestion.with-tooltip.openai', addSuggestion )
+				.on( 'click', '.translation-suggestion.with-tooltip.deepl', addSuggestion );
 		}
 	})( $gp.editor.install_hooks );
 
+	// Convert object to query params.
+	function convertObjectToQueryParam( object ) {
+		const params = new URLSearchParams();
+
+		Object.entries(object).forEach(([key, value]) => {
+		params.append(key, value);
+		});
+
+		return params.toString();
+	}
+
+	// Prefilter ajax requests to add translation_source to the request.
+	$.ajaxPrefilter( function ( options ) {
+		let data = Object.fromEntries( new URLSearchParams( options.data ) );
+
+		if ( 'POST' === options.type && $gp_editor_options.url === options.url && ! data.translation_source ) {
+
+			if ( $openAITranslationsUsed[ data.original_id ] ) {
+				data.translation_source = 'openai';
+				$openAITranslationsUsed.splice( [ data.original_id ] );
+			} 
+			if ( $deeplTranslationsUsed[ data.original_id ] ) {
+				data.translation_source = 'deepl';
```

Comment:
Let's not overwrite the source in preparation of storing the assist source separately.
```suggestion
				data.translation_assist = 'deepl';
```